accepted/1847 - FinalizationRegistry/proposal.md

```diff
@@ -247,15 +238,27 @@ abstract class Finalizable {
 typedef NativeFinalizer = Void Function(Pointer<Void>);
 typedef NativeFinalizerPtr = Pointer<NativeFunction<NativeFinalizer>>
 
-/// [FinalizationRegistry] which will execute its finalizers as early as
-/// possible without waiting for control to return to the event loop.
+/// [Finalizer] *may* call the given callback when objects it is attached to
+/// become inaccessible.
+///
+/// This callback is guaranteed to be called: when isolate group is shutting
+/// down for each object to which the finalizer is still attached.
 ///
-/// Will also invoke finalization callbacks when the isolate which created
-/// this finalization registry is shutting down.
-abstract class NativeFinalizationRegistry<F extends Finalizable>
-    extends FinalizationRegistry<Pointer> {
-  /// Creates a finalization registry with the given finalization
-  /// callback.
+/// No other promises are made that the callback will ever be called,
+/// only that *if* it is called with a finalization token as argument,
+/// at least one object registered in the registry with that finalization token
+/// is no longer accessible to the program.
+///
+/// If multiple finalizers has been attached to an object or the same
+/// finalizer has been attached multiple times to an object and that object
+/// becomes inaccessible to the program, then any number of those finalizers
+/// may trigger their associated callback. It will not necessarily be all or
+/// none of them.
+///
+/// Finalization callbacks will happen as early as possible, without waiting
+/// for control to return to the event loop.
+abstract class NativeFinalizer<T> {
+  /// Creates a finalizer with the given finalization callback.
   ///
   /// Note: [callback] is expected to be a native function which can be
```

Comment:
`[callback]` -> `The [callback]`

(because I don't like starting sentences with a lower-case letter, not even if it's code).